see/context/resources/qemu.py

```diff
@@ -282,13 +299,21 @@ def disk_clone(hypervisor, identifier, storage_pool, configuration):
       * backingStore/path if copy on write is enabled
 
     """
-    path = configuration['image']
-    cow = configuration['clone'].get('copy_on_write', False)
+    cow = configuration.get('copy_on_write', False)
 
     try:
-        volume = hypervisor.storageVolLookupByPath(path)
+        volume = hypervisor.storageVolLookupByPath(image)
     except libvirt.libvirtError:
-        raise RuntimeError("%s disk must be contained in a libvirt storage pool." % path)
+        if os.path.exists(image):
+            pool = hypervisor.storagePoolDefineXML(BASE_POOL_CONFIG.format(
+                os.path.dirname(image)))
+            pool.setAutostart(True)
+            pool.create()
+            pool.refresh()
```

Comment:
It is not redundant. This refreshes the newly created pool containing the golden image, so that cloning can happen, then line 374 refreshes the target pool. Line 374 also refreshes any other pool containing the working image if cloning is not requested.d